internal/stocker/aws_billing.go

```diff
@@ -1,71 +1,47 @@
 package stocker
 
 import (
-	"fmt"
 	"strconv"
 	"time"
 
+	cp "github.com/RHEcosystemAppEng/cluster-iq/internal/cloud_providers/aws"
 	"github.com/RHEcosystemAppEng/cluster-iq/internal/inventory"
 	"github.com/aws/aws-sdk-go/aws"
-	"github.com/aws/aws-sdk-go/aws/credentials"
-	"github.com/aws/aws-sdk-go/aws/session"
 	"github.com/aws/aws-sdk-go/service/costexplorer"
 	"go.uber.org/zap"
 )
 
 // AWSBillingStocker object to obtain costs and expenses from AWS Cost Explorer API
 type AWSBillingStocker struct {
-	region  string
+	// Account to scan on this stocker
 	Account *inventory.Account
-	logger  *zap.Logger
-	// AWS Session objects
-	apiSession          *session.Session
-	costExplorerSession *costexplorer.CostExplorer
-	// List of instances that needs expense updates
+	// Stocker Logger
+	logger *zap.Logger
+	// AWS connection interface
+	conn *cp.AWSConnection
+	// List of instances to obtain its expenses
 	Instances []inventory.Instance
 }
 
 // NewAWSStocker create and returns a pointer to a new AWSStocker instance
 func NewAWSBillingStocker(account *inventory.Account, logger *zap.Logger, instances []inventory.Instance) *AWSBillingStocker {
-	st := AWSBillingStocker{Account: account, region: defaultAWSRegion, logger: logger, Instances: instances}
-
-	// Creating Session for the AWS API
-	if err := st.Connect(); err != nil {
-		st.logger.Error("Failed to initialize new session during AWSBillingStocker creation", zap.Error(err))
+	// Leaving the region empty forces to the AWSConnection to use the default region until a new one is configured
+	conn, err := cp.NewAWSConnection(account.GetUser(), account.GetPassword(), "", logger, cp.WithCostExplorer())
```

Comment:
this logger should also be removed if it has been deleted from AWSConnectionOption.